modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/util/DTOConverterUtil.java

```diff
@@ -52,12 +62,51 @@ public static Object toDTO(
 					systemObjectDefinitionManager.getModelClassName());
 		}
 
+		return dtoConverter;
+	}
+
+	public static Object toDTO(
+			BaseModel<?> baseModel,
+			SystemObjectDefinitionManager systemObjectDefinitionManager,
+			User user)
+		throws Exception {
+
+		DTOConverter<BaseModel<?>, ?> dtoConverter = getDTOConverter(
+			systemObjectDefinitionManager);
+
+		Locale locale = null;
+
+		if (user != null) {
+			locale = user.getLocale();
+		}
+
+		DTOConverterRegistry dtoConverterRegistry =
+			_serviceTracker.getService();
+
 		DefaultDTOConverterContext defaultDTOConverterContext =
 			new DefaultDTOConverterContext(
 				false, Collections.emptyMap(), dtoConverterRegistry,
-				baseModel.getPrimaryKeyObj(), user.getLocale(), null, user);
+				baseModel.getPrimaryKeyObj(), locale, null, user);
 
 		return dtoConverter.toDTO(defaultDTOConverterContext);
 	}
 
+	@Activate
+	protected void activate(BundleContext bundleContext) {
+		_serviceTracker = ServiceTrackerFactory.open(
+			bundleContext.getBundle(), DTOConverterRegistry.class);
+	}
+
+	@Deactivate
+	protected void deactivate() {
+		if (_serviceTracker != null) {
+			_serviceTracker.close();
+
+			_serviceTracker = null;
+		}
+	}
+
+	private static ServiceTracker<DTOConverterRegistry, DTOConverterRegistry>
```

Comment:
Yes, that was my initial approach, but there was a rule in source formatter and I saw some jira tasks from core-infra to avoid using the `@Reference(unbind = "-")` code.

However, now I am testing it again and now it works. Maybe someone included an exception for this use case with static attributes.

Anyway, I will fixup the code with the commit

Thanks for bringing it up! :+1: